### ISSUE TITLE
Re10-react-quiz > src > components | StartScreen -> Question transition

### ISSUE THREAD

**Issue by devYuraKim** [devYuraKim/react#12]
I was expecting the React component to function like a traditional HTML page load or a Java controller, that clicking the button would navigate to the `Question` component. However, it doesn't work that way.

```
function StartScreen({ questions }) {
  return (
    <div className="start">
      <h2>Welcome to The React Quiz</h2>
      <h3>{questions.length} questions to test your React mastery</h3>
      <button className="btn btn-ui" onClick={???}>Let's start</button>
    </div>
  );
}

export default StartScreen;
```

**Comment by devYuraKim** [comment]
TL;DR: component rendering !== traditional page loading
---
Initially, I considered passing the Question component to the onClick handler, but realized that this approach wasn't feasible. I then thought about using a function that invokes the Question component, but that also didn't seem like the right solution.


**React components function differently from traditional HTML page loads or Java controllers**. 

While Router could be used for navigation, in this case, conditional rendering is used to display components based on state.

Based on the current `status` state, different components are rendered within the parent `App` component. 

In the `StartScreen` component, clicking the 'Let's Start' `button` triggers a `dispatch` function that updates the `status` from "`ready`" to "`active`". This status update satisfies the condition for rendering the `Question` component in the `App` component, creating an effect similar to a page transition.


---

**`StartScreen` component**
```
function StartScreen({ questions, dispatch }) {
  return (
    <div className="start">
      <h2>Welcome to The React Quiz</h2>
      <h3>{questions.length} questions to test your React mastery</h3>
      <button
        className="btn btn-ui"
        onClick={() => dispatch({ type: "dataActive" })}
      >
        Let's start
      </button>
    </div>
  );
}

export default StartScreen;
```


**`App` component**
```
return (
    <div className="app">
      <Header />
      <Main>
        {status === "loading" && <Loader />}
        {status === "error" && <Error />}
        {status === "ready" && (
          <StartScreen questions={questions} dispatch={dispatch} />
        )}
        {status === "active" && <Question />}
      </Main>
    </div>
  );
```
